src/SUMMARY.md

```diff
@@ -1,55 +1,58 @@
-  - [@std/trig](./std_trig.md)
-- [Examples with Explanations](./examples_with_explanations.md)
+- [About the documentation](./about_docs.md)
+- [Tutorials](./tutorials.md)
+  - [Getting Started](./getting_started.md)
   - [The many forms of "Hello, World!"](./many_forms_hello_world.md)
   - [More advanced examples](./advanced_examples.md)
-- [Transpile to Javascript](./transpile_js.md)
-- [Contributing to Alan](./contributing.md)
-- [Compiler Internals](./compiler_internals.md)
-  - [AMM Language Reference](./amm_language_reference.md)
-  - [AGA Language Reference](./aga_language_reference.md)
-  - [Alan Opcode Reference](./alan_opcode_reference.md)
\ No newline at end of file
+- [How-to Guides](./how_to.md)
+  - [Transpile to Javascript](./transpile_js.md)
+  - [Contribute to Alan](./contributing.md)
+- [Topic Guides](./topics.md)
+  - [About Alan](./about_alan.md)
+  - [Syntax and File Structure](./file_structure.md)
+    - [Implied Structure: Scope](./scope.md)
+    - [Constants](./constants.md)
+    - [Events](./events.md)
+    - [Functions](./functions.md)
+    - [Operators](./operators.md)
+    - [Statements](./statements.md)
+    - [Interfaces](./interfaces.md)
+    - [Handlers](./handlers.md)
+    - [User Defined Types](./user_types.md)
+    - [Error Handling](./error_handling.md)
+    - [Imports](./imports.md)
+    - [Exports](./exports.md)
+  - [Module System](./modules.md)
+    - [Definition](./module_definition.md)
+    - [Resolution](./module_resolution.md)
+    - [Acquisition](./module_acquisition.md)
+    - [Testing and Mocking](./module_testing.md)
+    - [Mutation and Permissioning](./module_mutation.md)
+- [Reference Guides](./reference.md)
+  - [List of Keywords and Reserved Symbols](./keywords.md)
+  - [Built-ins](./builtins/built_ins.md)
+    - [Built-in Types](./builtins/types.md)
+    - [Built-in Functions](./builtins/functions.md)
+      - [Type Coercion](./builtins/type_coercion.md)
+      - [Arithmetic](./builtins/arithmetic.md)
+      - [Logical and Bitwise](./builtins/bitwise.md)
+      - [Comparators](./builtins/comparators.md)
+      - [String Manipulation](./builtins/string_api.md)
+      - [Array Manipulation](./builtins/array_api.md)
+      - [HashMap Manipulation](./builtins/hashmap_api.md)
+      - ["Ternary" Functions](./builtins/ternary.md)
+      - [Error, Maybe, Result, and Either](./builtins/result_maybe.md)
+      - [Wait](./builtins/wait.md)
+    - [Built-in Operators](./builtins/operators.md)
+    - [Built-in Interfaces](./builtins/interfaces.md)
+  - [Standard Library](./standard_library.md)
+    - [@std/app](./std_app.md)
+    - [@std/seq](./std_seq.md)
+    - [@std/cmd](./std_cmd.md)
+    - [@std/deps](./std_deps.md)
+    - [@std/trig](./std_trig.md)
+  - [Compiler Internals](./compiler_internals.md)
+    - [AMM Language Reference](./amm_language_reference.md)
+    - [AGA Language Reference](./aga_language_reference.md)
+    - [Alan Opcode Reference](./alan_opcode_reference.md)
```

Comment:
This one is a bit interesting: does it belong in Reference Guides or Topic Guides? The Opcode reference surely belongs here, but the AMM and AGA references are a bit of a different beast, helping explain why along with how.

Reply:
I think this is fine for now, but maybe take the the extra explanations out and put them in their own topic guides?

Reply:
Yeah, I was torn on that one too. It could be in either. I wanted to keep the entire Compiler Internals section together and after reading all of the sections I went with this one. After the proposed tutorials and how-to guides are in place, I was planning/hoping we would update and beef up the reference and topics sections!